src/utils/test-utils.tsx

```diff
@@ -16,31 +22,42 @@ const goTo = (route?: string, name?: string) => {
     );
 };
 
-const waitForLoading = async () => {
-    return Promise.resolve();
-};
+// TODO - We use immer here but don't need it. The user metadata is currently just
+//  any number of keys (according to the types). Hoping that changes in the future.
+const generateMockUserMetadata = (username: string) => {
+    return produce(mockDeep<User['user_metadata']>(), (draft) => {
+        draft.avatar_url = `https://example.org/avatar/${username}`;
+        draft.email = `${username}@example.org`;
+        draft.email_verified = true;
+        draft.full_name = `Full ${username}`;
+        draft.iss = 'https://api.example.org';
+        draft.name = username;
+        draft.picture = `https://example.org/picture/${username}`;
+        draft.preferred_username = username;
+        draft.provider_id = '00000000000000000000_provider';
+        draft.sub = '00000000000000000000_sub';
+        draft.user_name = username;
 
-const MockUserProvider = ({ children, username }: any) => {
-    return (
-        <>
-            Your user name is {username} and {children}
-        </>
-    );
+        return draft;
+    });
 };
 
-const MockClientProvider = ({ children }: any) => {
+const MockProviders = ({ children, username }: any) => {
     const mockClient = createClient('http://mock.url', 'MockSupabaseAnonKey');
+    const mockAuthSession = mockDeep<AuthSession>();
 
-    const MockClient = createContext({});
+    if (username) {
+        mockAuthSession.user = mockDeep<User>();
+        mockAuthSession.user.user_metadata = generateMockUserMetadata(username);
+        Auth.useUser = () => mockAuthSession;
+    }
```

Comment:
Only need to mock out if a username is passed in. That's how we know if we should log the user in.